sky/backends/cloud_vm_ray_backend.py

```diff
@@ -409,6 +409,11 @@ def add_gang_scheduling_placement_group_and_setup(
 
         job_id = self.job_id
         if setup_cmd is not None:
+            if env_vars is None:
+                setup_envs = None
```

Comment:
In this case, are constants.SKYPILOT_NUM_NODES & other system env vars unavailable?